astroquery/nist/core.py

```diff
@@ -85,7 +85,13 @@ def _args_to_payload(self, *args, **kwargs):
 
         """
         request_payload = {}
-        request_payload["spectra"] = kwargs['linename']
+        linename = kwargs["linename"]
+        if isinstance(linename, str):
+            request_payload["spectra"] = linename
+        elif isinstance(linename, list):
+            request_payload["spectra"] = " ".join(linename)
+        else:
+            raise TypeError("linename must str or list")
```

Comment:
```suggestion
        request_payload["spectra"] = linename if isinstance(linename, str) else " ".join(linename)
```
And if `linename` cannot be handled this way then `str.join()` raises the `TypeError`.

Reply:
Thanks, this is a better catchall. The previous comment made me start thinking about this more broadly (lists, tuples, ndarrays, table columns, etc). I'll get it added in when I get a moment to later today.